### ISSUE TITLE
jax.random.truncated_normal generates -inf

### ISSUE THREAD

**Issue by david-berthelot** [google/jax#4559]
Values should be truncated  to `[-100, 100]` and most of the time it works. However for the seed in the example below it generates `-inf`.
```python
import jax.numpy as jn
import jax.random as jr
import jax
import jaxlib

jax.__version__
# '0.2.1'
jaxlib.__version__
# '0.1.55'

print(jr.truncated_normal(jn.array([41787198, 1040924135], 'uint32'), shape=(10000, 32), lower=-100, upper=100).min())
# -inf
```

**Comment by hawkinsp** [comment]
As it happens I fixed this yesterday: https://github.com/google/jax/pull/4548

It isn't in a jax release yet.
